workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectDispatcherSystem.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Gathers incoming dispatcher ops and invokes callbacks on relevant GameObjects.
+    /// </summary>
+    [UpdateInGroup(typeof(SpatialOSReceiveGroup.GameObjectReceiveGroup))]
+    public class GameObjectDispatcherSystem : ComponentSystem
+    {
+        public readonly Dictionary<int, SpatialOSBehaviourManager> EntityIndexToSpatialOSBehaviourManager = new Dictionary<int, SpatialOSBehaviourManager>();
+
+        public readonly HashSet<GameObjectComponentDispatcherBase> GameObjectComponentDispatchers =
```

Comment:
Why is this a `HashSet`? If we are iterating over it every loop, we want that to be as fast as possible, should be a list I think

Reply:
Good spot. Changed to a list.